pkg/server/sandbox_run_windows.go

```
@@ -54,13 +56,30 @@ func (c *criService) RunPodSandbox(ctx context.Context, r *runtime.RunPodSandbox
 	}
 	log.G(ctx).Debugf("Sandbox config %+v", config)
 
-	// Generate unique id and name for the sandbox and reserve the name.
-	id := util.GenerateID()
 	metadata := config.GetMetadata()
 	if metadata == nil {
 		return nil, errors.New("sandbox config must include metadata")
 	}
 	name := makeSandboxName(metadata)
+
+	// check if this is actually a restart
+	id, err := c.sandboxNameIndex.GetKey(name)
+	if err == nil && config.Annotations[annotations.AllowReset] == "true" {
+		log.G(ctx).Infof("reseting existing sandbox %q with id %q", name, id)
+		sandbox, err := c.sandboxStore.Get(id)
+		if err != nil {
+			return nil, errors.Wrapf(err, "an error occurred when try to find sandbox %q", id)
+		}
+		if err = c.resetSandbox(ctx, sandbox); err != nil {
+			return nil, errors.Wrapf(err, "could not reset sandbox %q", id)
+		}
+		return &runtime.RunPodSandboxResponse{PodSandboxId: id}, nil
+	} else if err != nil && err != registrar.ErrKeyNotReserved {
+		return nil, errors.Wrapf(err, "failed to check if sandbox name %q was previously reserved", name)
```

maybe we should just return the error as it is without wrapping. I wouldn't be surprised if there's logic somewhere that expects a particular error message.

---

Hopefully that code uses `errors.Is()`, but, regardless, I'll just log the error and let it fail on the reservation step, like you suggested

---

sounds good to me

---

just to clarify, I meant services which might not necessarily be written in Go and call into containerd via shell out, for example. But they probably wouldn't do exact string matches of output anyways.

---

Oh, I didnt think of that. The wrap should preserve the original error message, so as long as the match is via a `contains(err, "...")`, then it should be unaffected.